File: go/binary_search/count_test.go
package binary_search

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestCount_RandomCases(t *testing.T) {
	for range 10_000 {
		td := newRandomTestData()

		expected := td.expectedCount
		actual := CountInts(td.array, td.val)
		if actual != expected {
			assert.FailNowf(t, "",
				"CountInts(%v, %v):\n  Actual: %v\nExpected: %v",
				td.array, td.val, actual, expected,
			)
		}
	}
}

File: python/binary_search/count_test.py
from count import count_ints
from bs_random_test_data import RandomTestData


def test_random_cases():
	for _ in range(10_000):
		td = RandomTestData()

		expected = td.expected_count
		actual = count_ints(td.array, td.val)
		assert actual == expected, \
			f'count_ints({td.array}, {td.val}):\n  Actual: {actual}\nExpected: {expected}'
